### ISSUE TITLE
viz PlotTimeCourse

### ISSUE THREAD

**Issue by yishuiaoi** [CiaranWelsh/PyCoTools#9]
I am running the "Quick start" model in my jupyter lab notebook. I can simulate the model. When I visualized the results by` viz.PlotTimeCourse(TC, separate=False)`, I had an "AttributeError" as "'PlotTimeCourse' object has no attribute 'plot_kwargs'". 
How could I fix this to visualize the results?

**Comment by CiaranWelsh** [comment]
Hi Yishuiaoi, 

Unfortunately, I am unable to reproduce your problem. The code below works on my system. The only thing I can think of is maybe you are not using the most up to date version of PyCoTools. Please try reinstalling and trying again. To do this, use

```
$pip uninstall pycotools
$pip install pycotools --no-cache-dir
```
The `no-cache-dir` just ensures you do not use a cached version of pycotools, which it probably wont but no harm in being safe. Please let me know if this fixes your problem. 

All the best, 

Ciaran 

```
from pycotools import model, viz, tasks, misc
import os
import pandas
import seaborn
seaborn.set_style('white')
seaborn.set_context('notebook')

working_directory = r'/home/b3053674/Documents/Models/2018/01_Jan/PyCoToolsQuickStart'
copasi_file = os.path.join(working_directory, 'quick_start_model.cps')

if os.path.isfile(copasi_file):
    os.remove(copasi_file)

antimony_string = """
model michaelis_menten()
    compartment cell = 1.0
    var E in cell
    var S in cell
    var ES in cell
    var P in cell

    kf = 0.1
    kb = 1
    kcat = 0.3
    E = 75
    S = 1000

    SBindE: S + E => ES; kf*S*E
    ESUnbind: ES => S + E; kb*ES
    ProdForm: ES => P + E; kcat*ES
end
"""

with model.BuildAntimony(copasi_file) as loader:
    michaelis_menten = loader.load(antimony_string)

TC = tasks.TimeCourse(michaelis_menten, end=100,
                      step_size=1, intervals=100,
                      report_name='MM-time-course.csv')


viz.PlotTimeCourse(TC, separate=False, show=True)

```

**Comment by yishuiaoi** [comment]
I tried, and it did not work. 
I re-installed ANAconda, and did the steps as mentioned in the document. The same thing. 
One thing I want to mention, when I ran the "import code"
`import matplotlib
%matplotlib inline
from pycotools import model, viz, tasks, misc
import os
import pandas as pd
import numpy as np
import seaborn
seaborn.set_style('white')
seaborn.set_context('notebook')`

I received a lot of many warnings, I listed below. 

Does the last line of this warning matter?

> "from pandas.parser import CParserError"

> C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\_libs\__init__.py:4: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from .tslib import iNaT, NaT, Timestamp, Timedelta, OutOfBoundsDatetime
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\__init__.py:26: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import (hashtable as _hashtable,
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\dtypes\common.py:6: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import algos, lib
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\util\hashing.py:7: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import hashing, tslib
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\indexes\base.py:7: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import (lib, index as libindex, tslib as libts,
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\tseries\offsets.py:21: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  import pandas._libs.tslibs.offsets as liboffsets
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\ops.py:16: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import algos as libalgos, ops as libops
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\indexes\interval.py:32: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs.interval import (
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\internals.py:14: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import internals as libinternals
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\sparse\array.py:33: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  import pandas._libs.sparse as splib
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\window.py:36: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  import pandas._libs.window as _window
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\groupby\groupby.py:68: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import (lib, reduction,
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\core\reshape\reshape.py:30: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import algos as _algos, reshape as _reshape
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\io\parsers.py:45: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  import pandas._libs.parsers as parsers
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pandas\io\pytables.py:50: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected zd, got zd
  from pandas._libs import algos, lib, writers as libwriters
C:\Users\BME_IQ\Anaconda3\envs\py27\lib\site-packages\pycotools\viz.py:117: FutureWarning: The pandas.parser module is deprecated and will be removed in a future version. Please import from pandas.io.parser instead
  from pandas.parser import CParserError

**Comment by CiaranWelsh** [comment]
Hi Yishuiaoi,

As far as I can tell, the first set of warnings (from first to second-to-last) are a result of `pandas` importing `numpy`. Unfortunately I don't have much control over this right now, but they are benign, if a little unsightly. The last warning has been fixed and should be updated as I've just release a new version which is `1.0.8`. Please double check you are using the most up to date by using: 

    >>> import pycotools
    >>> pycotools.__version__

Now, as for the problem with plot_kwargs, I am quite confident that the code works as I have just tested it in a clean conda environment on both windows and linux. Here's what I did, 

1) create new conda env

    `$ conda create --name pycotools-test python-2.7`

2) switch to env (ignore `source` if on windows)

    `$ source activate pycotools-test`

3) install pycotools, making sure we first get rid of any old copies

    `$ pip uninstall pycotools -y`
    `$ pip install pycotools --no-cache-dir`

4) open an ipython shell (you may need to `pip install ipython`) by typing:

    `$ ipython`

5) Copy and paste the following and hit enter a few times. 

```
from pycotools import model, viz, tasks, misc
import os
import pandas
import seaborn
seaborn.set_style('white')
seaborn.set_context('notebook')

working_directory = os.getcwd()
copasi_file = os.path.join(working_directory, 'quick_start_model.cps')

if os.path.isfile(copasi_file):
    os.remove(copasi_file)

antimony_string = """
model michaelis_menten()
    compartment cell = 1.0
    var E in cell
    var S in cell
    var ES in cell
    var P in cell

    kf = 0.1
    kb = 1
    kcat = 0.3
    E = 75
    S = 1000

    SBindE: S + E => ES; kf*S*E
    ESUnbind: ES => S + E; kb*ES
    ProdForm: ES => P + E; kcat*ES
end
"""

with model.BuildAntimony(copasi_file) as loader:
    michaelis_menten = loader.load(antimony_string)

TC = tasks.TimeCourse(michaelis_menten, end=100,
                      step_size=1, intervals=100,
                      report_name='MM-time-course.csv')


viz.PlotTimeCourse(TC, separate=False, show=True)

```

Here's the output. 

![image](https://user-images.githubusercontent.com/19502680/44153934-5c6e5662-a0a1-11e8-8f29-e3e3f47051b8.png)

The only other thing I can think of is to ensure you are using a Jupyter python 2.7 kernel, and python 3 will not work. 

Good luck, and let me know how you get on, 

Ciaran

**Comment by yishuiaoi** [comment]
Hi, Ciaran,

Thank you very much. I did the script you provided. It works. I got the figure. 

A difference I observed this time compared with previous try. running the "pip install pycotools --no-cache-dir", I installed many related packages automatically. Previous, I installed the packages by installing them separately, as suggested in the "Document" website. 

It works fine, and thank you very much for helping me out. 
Yongliang
